### ISSUE TITLE
Unable to launch menu editor: 'cairo.Surface' conversion not found on 2.4.6

### ISSUE THREAD

**Issue by adjl** [linuxmint/Cinnamon#3850]
I'm running Linux Mint 17.1, Cinnamon 2.4.6.

I tried editing the menu by right-clicking on the Menu applet > Configure... > Open the menu editor, but the editor window didn't appear.

Next, I tried running it from the command line, but got the following error:
```
adjl lucensis ~ 
% cinnamon-menu-editor
Traceback (most recent call last):
  File "/usr/bin/cinnamon-menu-editor", line 19, in <module>
    main()
  File "/usr/bin/cinnamon-menu-editor", line 16, in main
    app.run()
  File "/usr/lib/cinnamon-menu-editor/cme/MainWindow.py", line 63, in run
    self.loadMenus()
  File "/usr/lib/cinnamon-menu-editor/cme/MainWindow.py", line 198, in loadMenus
    self.loadMenu({ None: None })
  File "/usr/lib/cinnamon-menu-editor/cme/MainWindow.py", line 213, in loadMenu
    icon = util.getIcon(menu, self.main_window)
  File "/usr/lib/cinnamon-menu-editor/cme/util.py", line 159, in getIcon
    wrapper.surface = Gdk.cairo_surface_create_from_pixbuf (pixbuf, widget.get_scale_factor(), widget.get_window())
TypeError: Couldn't find conversion for foreign struct 'cairo.Surface'
```

python-gi-cairo is already installed on my machine.

I tried the following:
- Update all packages (no effect)
- Restart Cinnamon (no effect)
- Restart machine (no effect)

This error is quite sudden, because the editor used to work for me. To the best that I know, I did not purposely remove packages that could cause this (apt-get update/autoremove, perhaps?).

Thanks

**Comment by mtwebster** [comment]
does `sudo apt-get install --reinstall cinnamon cinnamon-common` attempt to install anything extra?

**Comment by adjl** [comment]
It doesn't look like it.

```
adjl lucensis ~
% sudo apt-get install --reinstall cinnamon cinnamon-common
Reading package lists... Done
Building dependency tree        
Reading state information... Done
0 to upgrade, 0 to newly install, 2 reinstalled, 0 to remove and 7 not to upgrade.
Need to get 2,283 kB of archives.
After this operation, 0 B of additional disk space will be used.
Get:1 http://packages.linuxmint.com/ rebecca/main cinnamon-common all 2.4.6+rebecca [1,856 kB]
Get:2 http://packages.linuxmint.com/ rebecca/main cinnamon amd64 2.4.6+rebecca [427 kB]
Fetched 2,283 kB in 3s (718 kB/s) 
(Reading database ... 191104 files and directories currently installed.)
Preparing to unpack .../cinnamon-common_2.4.6+rebecca_all.deb ...
Unpacking cinnamon-common (2.4.6+rebecca) over (2.4.6+rebecca) ...
Preparing to unpack .../cinnamon_2.4.6+rebecca_amd64.deb ...
Unpacking cinnamon (2.4.6+rebecca) over (2.4.6+rebecca) ...
Processing triggers for libglib2.0-0:i386 (2.40.2-0ubuntu1) ...
Processing triggers for libglib2.0-0:amd64 (2.40.2-0ubuntu1) ...
Processing triggers for hicolor-icon-theme (0.13-1) ...
Processing triggers for man-db (2.6.7.1-1ubuntu1) ...
Processing triggers for mime-support (3.54ubuntu1.1) ...
Processing triggers for gnome-menus (3.10.1-0ubuntu2) ...
Processing triggers for desktop-file-utils (0.22-1ubuntu1) ...
Setting up cinnamon-common (2.4.6+rebecca) ...
Setting up cinnamon (2.4.6+rebecca) ...
Processing triggers for libc-bin (2.19-0ubuntu6.5) ...
```

**Comment by adjl** [comment]
I fixed the issue by reinstalling the following packages:
- libcairo-gobject2 (1.13.0~20140204-0ubuntu1.1)
- libcairo2 (1.13.0~20140204-0ubuntu1.1)
- python-cairo (1.8.8-1ubuntu5)
- python-gi (3.12.0-1ubuntu1)
- python-gobject (3.12.0-1ubuntu1)
- python-gobject-2 (2.28.6-12build1)
- python3-cairo (1.10.0+dfsg-3ubuntu2)
- python3-gi (3.12.0-1ubuntu1)

Note that I did so after upgrading from Linux Mint 17.2 to 17.3, Cinnamon 2.8.8; the upgrade itself did not resolve the issue.

That said, it appears the problem is not with Cinnamon, but by a faulty install. Whether the breakage was caused by Cinnamon in the first place is unclear, however...

Because I reinstalled them simultaneously, I cannot pinpoint the problematic package(s). :/

Cheers
